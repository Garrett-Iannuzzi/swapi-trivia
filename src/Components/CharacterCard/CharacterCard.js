import React from 'react';
import './CharacterCard.scss';
import PropTypes from 'prop-types';
import images from '../.././images/images.js';
import { Link } from 'react-router-dom';

const CharacterCard = ({ character, creature, name, openingCrawl, population, relatedFilms, species, resetCharacters, handleFavorites }) => {
  const films = relatedFilms.map((film, index) => <p className='h2-relatedFilm' key={index}>{film.relatedFilms}</p>);
  const characterInfo = {
    character: character,
    creature: creature,
    name: name,
    population: population,
    relatedFilms: relatedFilms,
    species: species
  }
  return(
    <article className='character-card'>
      <header className='header-character'>
        <h2 className='h2-character-header'>{character}</h2>
        <button className='button-favorites' onClick={ () => {handleFavorites(characterInfo)} }>
          <img className='favorites-img' src={images.heart} alt='heart icon for favorites' />
        </button>
      </header>
      <section className='section-character'>
        <h2 className='h2-character'>Species: {species}</h2>
        <h2 className='h2-character'>Creature: {creature}</h2>
        <h2 className='h2-character'>Name: {name}</h2>
        <h2 className='h2-character-population'>Population: {population}</h2>
        <h2 className='h2-films'>Films: </h2>
            {films}
      </section>
      <Link to='/movies' className='link-back-movies'>
        <button className='button-movies' onClick={() => resetCharacters()}>Return to Movies</button>
      </Link>
    </article>
  )

}

export default CharacterCard;

CharacterCard.propTypes = {
  character: PropTypes.string,
  creature: PropTypes.string,
  name: PropTypes.string,
  openingCrawl: PropTypes.string,
  population: PropTypes.string,
  relatedFilms: PropTypes.arrayOf(PropTypes.object),
  species: PropTypes.string,
  characterInfo: PropTypes.func
}
