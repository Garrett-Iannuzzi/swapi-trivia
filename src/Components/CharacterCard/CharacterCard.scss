
.character-card {
  border: solid 3px white;
  border-radius: 10px;
  background-color: black;
  color: white;
  display: flex;
  flex-direction: column;
  height: 375px;
  align-items: center;
  margin: 10px;
  opacity: .9;
  width: 30%;
  min-width: 400px;
}

.favorites-img {
  height: 30px;
  width: 30px;
}

.favorites-img:hover {
  transform: scale(1.1);
}

.button-favorites {
  background-color: black;
  opacity: .9;
  border: 0;
  display: flex;
  align-items: center;
  margin-right: 15px;
  cursor: pointer;
}

.header-character {
  height: 50px;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.section-character {
  display: flex;
  height: 325px;
  align-items: center;
  overflow: scroll;
}

.h2-character {
  width: 100%;
  text-align: center;
}
