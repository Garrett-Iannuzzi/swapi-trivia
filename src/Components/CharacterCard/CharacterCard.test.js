import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import CharacterCard from '../CharacterCard/CharacterCard.js';

describe('CharacterCard', () => {
  let wrapper;
  const mockHandleFavorites = jest.fn();
  const resetCharactersMock = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<CharacterCard
      character="C-3PO"
      creature="artificial"
      name="Tatooine"
      openingCrawl="the guardians of"
      population="200000"
      relatedFilms={[
        {relatedFilms: "The Empire Strikes Back"},
        {relatedFilms: "Attack of the Clones"}]}
      species="Droid"
      handleFavorites={mockHandleFavorites}
      resetCharacters={resetCharactersMock}
    />);
  })

  it('Should match the snapshot with data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should call handleFavorites with characterInfo on click', () => {
    const characterInfo = {
      character: "C-3PO",
      creature: "artificial",
      name: "Tatooine",
      population: "200000",
      relatedFilms: [
        {relatedFilms: "The Empire Strikes Back"},
        {relatedFilms: "Attack of the Clones"}],
      species: "Droid"
    }
    wrapper.find('.button-favorites').simulate('click');
    expect(mockHandleFavorites).toHaveBeenCalledWith(characterInfo);
  });

  it('should call resetCharacters on click', () => {
    wrapper.find('.button-movies').simulate('click');
    expect(resetCharactersMock).toHaveBeenCalled();
  });
});
