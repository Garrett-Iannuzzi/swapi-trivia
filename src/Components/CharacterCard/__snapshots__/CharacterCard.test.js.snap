// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard Should match the snapshot with data passed in 1`] = `
<article
  className="character-card"
>
  <header
    className="header-character"
  >
    <h2>
      C-3PO
    </h2>
    <button
      className="button-favorites"
      onClick={[Function]}
    >
      <img
        alt="heart icon for favorites"
        className="favorites-img"
        src="lover.png"
      />
    </button>
  </header>
  <section
    className="section-character"
  >
    <h2
      className="h2-character"
    >
      Droid
    </h2>
    <h2
      className="h2-character"
    >
      artificial
    </h2>
    <h2
      className="h2-character"
    >
      Tatooine
    </h2>
    <h2
      className="h2-character"
    >
      200000
    </h2>
    <h2
      className="h2-character"
      key="0"
    >
      The Empire Strikes Back
    </h2>
    <h2
      className="h2-character"
      key="1"
    >
      Attack of the Clones
    </h2>
  </section>
</article>
`;
